File: 365/365.c
#include "stdbool.h"
#include "stdio.h"

int _GCD(int a, int b) {
    /* while(a != 0 && b != 0) { */
    /*     int tmp = a; */
    /*     a = b % a; */
    /*     b = tmp; */
    /* } */
    /* return a > b ? a : b; */
    while (b) {
        b ^= a ^= (b ^= (a %= b));
    }
    return a;
}

bool canMeasureWater(int jug1Capacity, int jug2Capacity, int targetCapacity){
    return targetCapacity <= jug1Capacity + jug2Capacity &&
        targetCapacity % _GCD(jug1Capacity, jug2Capacity) == 0;
}

int main(int argc, char *argv[]) {
    int jug1Capacity, jug2Capacity, targetCapacity;
    fscanf(stdin, "%d %d %d", &jug1Capacity, &jug2Capacity, &targetCapacity);
    fprintf(stdout, "%d", canMeasureWater(jug1Capacity, jug2Capacity, targetCapacity));
    return 0;
}

File: 365/365.rs
struct Solution {}

impl Solution {
    fn _gcd(a: i32, b: i32) -> i32 {
        let mut _a = a;
        let mut _b = b;
        while _b != 0 {
            let _tmp: i32 = _b;
            _b = _a % _b;
            _a = _tmp;
        }
        return _a;
    }
    pub fn can_measure_water(jug1_capacity: i32, jug2_capacity: i32, target_capacity: i32) -> bool {
        return jug1_capacity + jug2_capacity >= target_capacity &&
            target_capacity % Solution::_gcd(jug1_capacity, jug2_capacity) == 0;
    }
}

fn main() {
    println!("{}", Solution::can_measure_water(7, 5, 3));
}
